Documentation/articles/content/why_content_pipeline.md

```diff
@@ -0,0 +1,59 @@
+
+Most platforms have audio processing that is optimized to certain compressed formats. By not using them you loose performance and system memory.  Often this is when people resort to just using a format optimized for your current target platform. You could save the sound effects as ADPCM and run pretty optimally on Windows systems. The problem here is only apparent once you try to take your game to another platform where ACPCM is not optimal. You need to manually re-export all your game content in the new optimal format for that platform… and you did keep your original content uncompressed right?
+
+## Shaders
+
+This is an area that causes real pain, custom shaders. There are a number of shading languages that you can use depending on the platform you are targeting. For OpenGL based systems that is GLSL, for DirectX based systems it is HLSL, there is also CG from Nvidia.
+
+The Effect system in XNA/MonoGame was designed around the HLSL language as it is based around the .fx format, which allows a developer to write both vertex and pixel shaders in one place. Historically both GLSL and HLSL have separate vertex and pixel shaders.  HLSL until recently compiled and linked these at build time, however, GLSL does this at runtime. Now without a content pipeline or some form of tooling a developer would need to write two shaders, one for HLSL and one for GLSL. The good news is the MonoGame MGFX.exe tool can create a shader in .fx format and also enable it to work in GLSL. It does this by using an open-source library called libmojoshader, which does some funky HLSL to GLSL instruction conversion to create an OpenGL-based shader, but rather than doing that at runtime, we do it at build time so we do not need to deploy mojoshader with the OpenGL based games. All this saves you the hassle of having to write and maintain two shaders.
+
+> From MonoGame 3.8, the MGFX tool can build content on any platform. For MonoGame 3.7 or earlier, MGFX only runs on a windows box, this is because it needs the DirectX shader tooling to compile the HLSL before passing it to libmojoshader for conversion (for OpenGL platform targets).
+
+## Models
+
+For the most part, the model support in XNA/MonoGame is pretty good. XNA supports .x, .fbx files for 3D models.  MonoGame, thanks to the excellent assimp project, supports a much wider range of models including .3ds. However, some of these formats might produce some weirdness at render time as only .fbx has been fully tested. Also note that assimp does not support the older version .fbx format which ships with most of the original XNA samples, so you will need to [convert those to the new format manually](https://www.codeproject.com/articles/1041397/updating-old-fbx-files-for-the-modern-era).
```

Comment:
> For the most part, the model support in XNA/MonoGame is pretty good.

This is all sounding too much like a personal blog post and less like documentation.

The OpenGL docs don't talk about "well threading sort of works".  I would remove personal anecdotes from these docs and focus on the feature set.

Reply:
It was a blog, as it was @dellis1972 , although it did have a fair bit of revision already.